src/Workspaces/Core/Portable/Options/EditorConfig/EditorConfigOptionsGenerator.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using Microsoft.CodeAnalysis.CodeStyle;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.PooledObjects;
+
+namespace Microsoft.CodeAnalysis.Options.EditorConfig
+{
+    [Export(typeof(EditorConfigOptionsGenerator)), Shared]
+    internal class EditorConfigOptionsGenerator
+    {
+        private readonly IEnumerable<Lazy<IEditorConfigOptionsCollection, LanguageMetadata>> _editorConfigGenerators;
+
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public EditorConfigOptionsGenerator(
+            [ImportMany] IEnumerable<Lazy<IEditorConfigOptionsCollection, LanguageMetadata>> generators)
+        {
+            _editorConfigGenerators = generators;
+        }
+
+        public ImmutableArray<(string feature, ImmutableArray<IOption2> options)> GetDefaultOptions(string language)
+        {
+            var builder = ArrayBuilder<(string, ImmutableArray<IOption2>)>.GetInstance();
+            builder.AddRange(GetLanguageAgnosticEditorConfigOptions());
+
+            var generators = _editorConfigGenerators.Where(b => b.Metadata.Language == language);
+            foreach (var generator in generators)
+            {
+                builder.AddRange(generator.Value.GetLanguageSpecificEditorConfigOptions());
+            }
+
+            return builder.ToImmutableAndFree();
```

Comment:
This no longer returns default that caused some concern. However, if the user provides an invalid string (e.g. "CSharp" instead of "C#" this will fail silently by returning the language agnostic options. 

Should I provide some sort of warning if the language name provided doesn't match anything?